BlueMapCore/src/main/java/de/bluecolored/bluemap/core/threejs/BufferGeometry.java

```diff
@@ -127,6 +131,109 @@ public String toJson() {
 		}
 	}
 
+	public byte[] toBinary() {
+		byte[][] encodedAttrs = new byte[this.attributes.size()][];
+
+		int[] groupArray = new int[this.groups.length * 3];
+		int i = 0;
+		for (MaterialGroup g : this.groups) {
+			groupArray[i++] = g.getMaterialIndex();
+			groupArray[i++] = g.getStart();
+			groupArray[i++] = g.getCount();
+		}
+		ByteBuffer buffer = ByteBuffer.allocate(4 * groupArray.length);
+		IntBuffer intBuffer = buffer.asIntBuffer();
+		intBuffer.put(groupArray);
+		byte[] encodedGroups = buffer.array();
+
+		int[] header = new int[this.attributes.size() * 5 + 2];
+		int current = 12 + header.length * 4;
+		header[0] = current;
+		header[1] = encodedGroups.length;
+		current +=  encodedGroups.length;
+		int j = 2;
+		for (Entry<String, BufferAttribute> entry : this.attributes.entrySet()) {
+			header[j ++] = current;
+			byte[] data = entry.getValue().toBinary();
+			header[j ++] = data.length;
+			current += data.length;
+			int type = AttributeType.valueOf(entry.getKey()).ordinal();
+			header[j ++] = type;
+			encodedAttrs[type] = data;
+			header[j ++] = entry.getValue().isNormalized()?1:0;
+			header[j ++] = entry.getValue().getItemSize();
+		}
```

Comment:
Is there something speaking against moving the `isNormalized` and `itemSize` properties into the `toBinary()` method of the buffer-attribute? To me this would seem more logical.